internal/provider/bootstrap.go

```diff
@@ -48,9 +48,11 @@ func Bootstrap(e *pluggable.Event) pluggable.EventResponse {
 	}
 	// TODO: this belong to a systemd service that is started instead
 
-	tokenNotDefined := (providerConfig.Kairos != nil && providerConfig.Kairos.NetworkToken == "")
+	p2pBlockDefined := providerConfig.P2P != nil
+	tokenNotDefined := ((p2pBlockDefined && providerConfig.P2P.NetworkToken == "") || !p2pBlockDefined)
+	skipAuto := (p2pBlockDefined && !providerConfig.P2P.Auto.IsEnabled())
```

Comment:
I think it has to be:
```suggestion
	skipAuto := !(p2pBlockDefined && providerConfig.P2P.Auto.IsEnabled())
```
because if `p2pBlockDefined` is `false`, we want skipAuto to be `true`, right?

Reply:
I think it's equivalent, skipAuto is just used to check the fields instead of the whole logic, we check down the line if the user have enabled any `k3s` block